keras_cv/layers/preprocessing/random_height.py

```diff
@@ -0,0 +1,129 @@
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+from keras_cv.utils import preprocessing
+
+BILINEAR = "bilinear"
+NEAREST = "nearest"
+BICUBIC = "bicubic"
+AREA = "area"
+LANCZOS3 = "lanczos3"
+GAUSSIAN = "gaussian"
+MITCHELLCUBIC = "mitchellcubic"
+HEIGHT_AXIS = -3
+WIDTH_AXIS = -2
+IMAGES = "images"
+BOUNDING_BOXES = "bounding_boxes"
+
+class RandomHeight(BaseImageAugmentationLayer):
+    """A preprocessing layer which randomly varies image height during training.
+    This layer adjusts the height of a batch of images by a random factor.
+    The input should be a 3D (unbatched) or 4D (batched) tensor in the
```

Comment:
maybe add detail as to the behavior for intra-batch? Is is all samples within a batch using the same height / width, or randomness is applied intra-batch?

Reply:
Does this imply to use ragged tensor if not padded?

Reply:
Alright, shall do

Reply:
> Does this imply to use ragged tensor if not padded?

I think it might, because how else would the different tensor configurations be handled?